### PR TITLE
BREAKING CHANGE/sqlalchemy instead of peewee

### PR DESCRIPTION
**Pull request by jonathan-rohde**
# Pull Request Checklist

### Note to first-time contributors: Please open a discussion post in [Discussions](https://github.com/open-webui/open-webui/discussions) and describe your changes before submitting a pull request.

**Before submitting, make sure you've checked the following:**

- [x] **Target branch:** Please verify that the pull request targets the `dev` branch.
- [x] **Description:** Provide a concise description of the changes made in this pull request.
- [x] **Changelog:** Ensure a changelog entry following the format of [Keep a Changelog](https://keepachangelog.com/) is added at the bottom of the PR description.
- [ ] **Documentation:** Have you updated relevant documentation [Open WebUI Docs](https://github.com/open-webui/docs), or other documentation sources?
- [ ] **Dependencies:** Are there any new dependencies? Have you updated the dependency versions in the documentation?
- [x] **Testing:** Have you written and run sufficient tests for validating the changes?
- [x] **Code review:** Have you performed a self-review of your code, addressing any coding standard issues and ensuring adherence to the project's coding standards?
- [x] **Prefix:** To cleary categorize this pull request, prefix the pull request title, using one of the following:
  - **BREAKING CHANGE**: Significant changes that may affect compatibility
  - **build**: Changes that affect the build system or external dependencies
  - **ci**: Changes to our continuous integration processes or workflows
  - **chore**: Refactor, cleanup, or other non-functional code changes
  - **docs**: Documentation update or addition
  - **feat**: Introduces a new feature or enhancement to the codebase
  - **fix**: Bug fix or error correction
  - **i18n**: Internationalization or localization changes
  - **perf**: Performance improvement
  - **refactor**: Code restructuring for better maintainability, readability, or scalability
  - **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)
  - **test**: Adding missing tests or correcting existing tests
  - **WIP**: Work in progress, a temporary label for incomplete or ongoing work

# Changelog Entry

### Description

Replacing peewee with SQLAlchemy. The combination of FastAPI with Peewee is not supported and we should avoid using both libraries in parallel.
The change replaces the whole ORM model with SQLAlchemy and uses Alembic for schema migrations.
The initial database migration script is checking the database schema for table existence and skips table creation in case the table already exists.

### Changed

- Replace Peewee with SQLAlchemy
- Replace peewee-migrate with alembic

### Breaking Change

- The fact that there might be columns added between installed versions is making the replacement of the migration a breaking change.

---

### Additional Information

- Related to issue #3228

### CODE REVIEW

**Comment by justinh-rahb** [comment]
We'll still need to investigate this matter in regards to the SQL connections patch:
https://github.com/open-webui/open-webui/actions/runs/9597080436/job/26465379431?pr=3323#step:9:1

**Comment by MichaelJThomas-2016** [comment]
Just commenting, but you're doing exactly what I was thinking of doing :).

**Comment by perfectra1n** [comment]
Yeah, I should've changed my PR to use `/health` instead of `/api/tags` (that endpoint doesn't even exist, it's supposed to be `/api/v1/tags/all` anyways).

File: backend/apps/ollama/main.py [hunk 1]
```diff
@@ -711,6 +712,7 @@ async def generate_chat_completion(
     form_data: GenerateChatCompletionForm,
     url_idx: Optional[int] = None,
     user=Depends(get_verified_user),
+    db=Depends(get_db),
```
Comment (tjbck) [inline backend/apps/ollama/main.py:715]: Is there a workaround for this?

File: backend/apps/ollama/main.py [hunk 2]
```diff
@@ -724,7 +726,7 @@ async def generate_chat_completion(
     }
 
     model_id = form_data.model
-    model_info = Models.get_model_by_id(model_id)
+    model_info = Models.get_model_by_id(db, model_id)
```
Comment (tjbck) [inline backend/apps/ollama/main.py:729]: Ideally we wouldn't want to modify the router codes to include the db param if possible.

**Comment by tjbck** [comment]
Promising start! I just wish there is a workaround for adding session db param for every db call we make, I hope it's an engineering issue we can fix 🙌😅

**Comment by jonathan-rohde** [comment]
Made some more changes. Now the router does not need changes to the parameters

File: src/lib/apis/models/index.ts [hunk 3]
```diff
@@ -63,10 +63,7 @@ export const getModelInfos = async (token: string = '') => {
 export const getModelById = async (token: string, id: string) => {
 	let error = null;
 
-	const searchParams = new URLSearchParams();
```
Comment (tjbck) [inline src/lib/apis/models/index.ts:66]: Mistake here? I don't see why this was changed?

File: backend/apps/webui/routers/models.py [hunk 4]
```diff
@@ -53,7 +56,7 @@ async def add_new_model(
 ############################
 
 
-@router.get("/", response_model=Optional[ModelModel])
```
Comment (tjbck) [inline backend/apps/webui/routers/models.py:56]: This fails if the `id` includes `/`. This should not have been changed.

File: src/lib/apis/models/index.ts (hunk 3 above)
Comment (jonathan-rohde) [reply src/lib/apis/models/index.ts]: Yes you are right. The change was based on a integration test in backend which was not able to get the model by id. Looked like a conflict in the router with same path but different query parameters.
Reverted the change

**Comment by tjbck** [comment]
Merging this to interim 'dev-migration' branch for now, Thanks for all the hard work! 🙌

**Pull request merged**